pywbem_mock/_wbemconnection_mock.py

```diff
@@ -119,6 +110,244 @@ def _cvt_obj_name(objname):
     return objname
 
 
+class ProviderRegistry(object):
+    """
+    This class defines the provider registry with methods to register
+    a provider and to get the registered provider for a particular classname
+    namespace, and provider_type.
+    """
+    def __init__(self):
+        self._registry = NocaseDict()
+
+    def __repr__(self):
+        return _format(
+            "ProviderRegistry("
+            "registry={s._registry}, ",
+            s=self)
+
+    def register_provider(self, conn, provider, namespaces=None,
+                          schema_pragma_files=None, verbose=None):
+        # pylint: disable=line-too-long
+        """
+        Register the provider object for specific namespaces and CIM classes.
+        Registering a provider tells the FakedWBEMConnection that the provider
+        implementation provided with this call as the `provider` parameter is
+        to be executed as the request response method for the namespaces
+        defined in the `namespaces` parameter, the provider type defined in the
+        provider 'provider_type` attribute of the `provider` and the classes
+        defined in the provider `provider_classnames` attribute of the
+        `provider`.
+
+        The provider registration process includes:
+
+        1. Validation that the namespaces defined for the provider exist.
+        2. Validation that the superclass of the provider is consistent with
+           the `provider_type` attribute defined in the provider.
+        3. Installation of any CIM classes defined by the provider
+           (`provider_classnames` attribute) including installation of
+           dependencies for these classes using the `schema_pragma_files` to
+           locate the search directories for dependencies.
+        4. Adding the provider to the registry of user_providers so that any
+           of the request methods defined for the `provider_type` are
+           passed to this provider in place of the default request processors.
+
+        Providers can only be registered for the following request response
+        methods:
+
+        1. provider_type = 'instance': CreateInstance, ModifyInstance, and
+           DeleteInstance
+
+        2. provider_type = 'method': InvokeMethod.
+
+        Each classname in a particular namespace may have at most one provider
+        registered
+
+        Parameters:
+
+          conn  (:class:`~pywbem_mock.FakedWBEMConnection`):
+            Defines the attributes of the connection. Used to issue requests to
+            create instances in the interop namespace for all existing
+            namespaces that do not have instances of CIM_Namespace defined
+
+          provider (instance of subclass of :class:`pywbem_mock:InstanceWriteProvider` or :class:`pywbem_mock:MethodProvider`):
```

Comment:
Change colon between pywbem_mock and class name to a dot.

Reply:
There were two of them.  DONE